src/main/thrift/parquet.thrift

```diff
@@ -547,6 +547,61 @@ struct RowGroup {
   4: optional list<SortingColumn> sorting_columns
 }
 
+/** Identifier for built-in sort order used to produce min and max values. */
+enum Order {
+  /**
+   * The signed ordering is the order produced by comparing single primitive
+   * values with a signed comparator, or the lexicographic ordering produced by
+   * comparing each byte of a binary or fixed using a signed comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  SIGNED = 0;
+
+  /**
+   * The unsigned ordering is produced by comparing single primitive values
+   * with an unsigned comparison, or the lexicographic ordering produced by
+   * comparing each byte of a binary or fixed using an unsigned comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  UNSIGNED = 1;
+
+  /**
+   * Identifiers for custom orderings, to be defined in the ColumnOrder struct.
+   */
+  CUSTOM = 2;
+}
+
+/** Descriptor for the order used for min, max, and sorting values in a column
+ */
+struct ColumnOrder {
+  /** The order used for this column */
+  0: required Order order;
+
+  /** Whether the order used is ascending (true) or descending (false) */
+  1: required boolean is_ascending;
+
+  /**
+   * A string that identifies the order for this column. This field should be
+   * set if the order is any value other than SIGNED or UNSIGNED and is used to
+   * identify the actual order used for min, max, and soring values.
+   *
+   * This identifier should follow one of the following formats:
+   * * 'icu54:<locale-keyword>' - ICU 54 ordering for the ICU 54 locale keyword
```

Comment:
Since adding an enum value is a breaking change in thrift and ordering is ignorable by old libraries that don't understand it, I'd favor a model like the one proposed by Alex above.
We need to verify that the union type implementation in thrift doesn't suffer from the same issue. A quick glance at the code seems to confirm it's fine but we just need to try it. 
In that case I think `1: required Order order` needs to be optional since it would be null when reading an unknown type.

If Union doesn't work, another approach (example bellow) is to simply have optional fields which we document as exclusive. However Union is cleaner.
```
struct ColumnOrder {
  // only one should be set
  1: optional boolean signed; // whether default signed unsigned binary comparison is used
  2: optional string sql_collation; // using sql collation string 
  3: optional string icu_54_locale; // using unicode locale
}
```